tests/e2e/docker/initialize.sh

```diff
@@ -5,3 +5,6 @@ echo "Initializing WooCommerce E2E"
 wp plugin install woocommerce --activate
 wp theme install twentynineteen --activate
 wp user create customer customer@woocommercecoree2etestsuite.com --user_pass=password --role=customer --path=/var/www/html
+
+# we cannot create API keys for the API, so we using basic auth, this plugin allows that.
```

Comment:
In fact we can create API keys for our own REST API without having to install this plugin, unless you are planning to do some requests to WP endpoints instead of WooCommerce core endpoints.

Reply:
Initially I had planned on using the API authentication and even [wrote an interceptor for it](https://github.com/woocommerce/woocommerce/blob/e4d90f10c3643dc56d003384b407979ad17c21d4/tests/e2e/factories/src/framework/api/axios/axios-oauth-interceptor.ts)! The idea was that we would store a `consumer_key` and `consumer_secret` in the test suite config file and then generate that key when initializing the test suite. 
 Creating a new CLI command felt a little outside the scope of this PR though and so the basic auth plugin provided by WordPress seemed like a reasonable way to quickly get this working. 

I'll note that I did create an API key on my local site though and thoroughly test the interceptor so if we decide to go that route it will be a very easy change to make :smile: